spoj_practice/spoj2/spoj2.py

```diff
@@ -7,43 +8,29 @@
 # N < 1,000,000,000
 # Find all primes between a given n, m
 
-
 # handle stdin/stdout and unit testing
 
-
-
 # returns a list of all the primes in between
 def find_primes_between (n, m):
     # divide by all odd numbers, counter it up
     # divide by all primes before it
     # divide by all other primes between n, m
     # make sure we start with an odd number
-    
+
     # primes to check against
     # 1 is not a prime number, the loop excludes even numbers thus 2 does not need to be added
-    primes = [3]
-    primes_between = []
-    
-    # unique cases
-    if m < 2:
-        primes_between.append(2)
-    if m <= 3:
-        primes_between.append(3)
-    
+    primes = [2, 3]
```

Comment:
Are you really moduloing by a list though? E.g. if 2 was in the list, are
you doing 2 % [2]?

What happens when you iterate over an empty list? Do you get None? Try it

Try running your code without 2 in the list. Does it work?
 On Apr 8, 2015 11:00 AM, "Kevin Lau" <notifications@github.com> wrote:

> In spoj_practice/spoj2/spoj2.py
> <https://github.com/thekevlau/interview_training/pull/2#discussion_r27995688>
> :
>
> >      # primes to check against
> >      # 1 is not a prime number, the loop excludes even numbers thus 2 does not need to be added
> > -    primes = [3]
> > -    primes_between = []
> > -
> > -    # unique cases
> > -    if m < 2:
> > -        primes_between.append(2)
> > -    if m <= 3:
> > -        primes_between.append(3)
> > -
> > +    primes = [2, 3]
>
> because we need to start the list of primes at something, the logic is
> that we divide by all known primes, thus there must be at least one known
> prime to divide by when going in, otherwise what are you modulo-ing by? an
> empty list? don't think that's kosher....what would that evaluate to, 2 %
> None ? that returns a type error. if it's 2 % [], the interpreter returns
> unsupported operand type since you're comparing an int and a list
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/thekevlau/interview_training/pull/2/files#r27995688>.
>